packages/office-ui-fabric-react/src/components/ContextualMenu/ContextualMenu.classNames.ts

```diff
@@ -1,28 +1,123 @@
 import { memoizeFunction } from '../../Utilities';
-import { mergeStyleSets, ITheme } from '../../Styling';
-import { IContextualMenuStyles } from './ContextualMenu.Props';
-
+import { ITheme, mergeStyles } from '../../Styling';
+import { IContextualMenuStyles, IMenuItemStyles } from './ContextualMenu.Props';
+import { getStyles as getContextualMenuStyles, getMenuItemStyles } from './ContextualMenu.styles';
 export interface IContextualMenuClassNames {
+  container?: string;
+  root?: string;
+  list?: string;
+  divider?: string;
+  header?: string;
   title?: string;
 }
 
-export const getClassNames = memoizeFunction((
-  styles: IContextualMenuStyles,
+export interface IMenuItemClassNames {
+  item?: string;
+  root?: string;
+  linkContent?: string;
+  icon?: string;
+  subMenuIcon?: string;
+  label?: string;
+}
+
+export const getContextualMenuClassNames = memoizeFunction((
   theme: ITheme,
+  className: string
 ): IContextualMenuClassNames => {
 
-  styles = styles || {};
+  const styles = getContextualMenuStyles(theme);
 
-  return mergeStyleSets({
-    title: [{
-      fontSize: '16px',
-      paddingRight: '14px',
-      paddingLeft: '14px',
-      paddingBottom: '5px',
-      paddingTop: '5px',
-      backgroundColor: theme.palette.neutralLight
-    },
-    styles.title
-    ]
-  });
+  return {
+    container: mergeStyles(
```

Comment:
Another thing you can do is call mergeStyleSets once instead of merge styles everywhere. You can do 
```typescript
mergeStyleSets({
 container: ['ms-contextualMenu-container', className, styles.container],
 root['ms-ContextialMenu is-open', styles.root']
})
```